src/Analyzers/AnalyzerRunner.cs

```diff
@@ -25,28 +24,33 @@ internal partial class AnalyzerRunner : IAnalyzerRunner
             CodeAnalysisResult result,
             ImmutableArray<DiagnosticAnalyzer> analyzers,
             Project project,
-            ImmutableArray<string> formattableDocumentPaths,
+            ImmutableHashSet<string> formattableDocumentPaths,
             ILogger logger,
             CancellationToken cancellationToken)
         {
-            var compilation = await project.GetCompilationAsync(cancellationToken);
+            var compilation = await project.GetCompilationAsync(cancellationToken).ConfigureAwait(false);
             if (compilation is null)
             {
                 return;
             }
 
-            var analyzerCompilation = compilation.WithAnalyzers(
-                analyzers,
-                options: project.AnalyzerOptions,
-                cancellationToken);
-            var diagnostics = await analyzerCompilation.GetAnalyzerDiagnosticsAsync(cancellationToken);
+            var analyzerOptions = new CompilationWithAnalyzersOptions(
+                project.AnalyzerOptions,
+                onAnalyzerException: null,
+                concurrentAnalysis: true,
+                logAnalyzerExecutionTime: false,
+                reportSuppressedDiagnostics: false);
+            var analyzerCompilation = compilation.WithAnalyzers(analyzers, analyzerOptions);
+            var diagnostics = await analyzerCompilation.GetAnalyzerDiagnosticsAsync(cancellationToken).ConfigureAwait(false);
+
             // filter diagnostics
             foreach (var diagnostic in diagnostics)
             {
                 if (!diagnostic.IsSuppressed &&
                     diagnostic.Severity >= DiagnosticSeverity.Warning &&
                     diagnostic.Location.IsInSource &&
-                    formattableDocumentPaths.Contains(diagnostic.Location.SourceTree?.FilePath, StringComparer.OrdinalIgnoreCase))
+                    diagnostic.Location.SourceTree is object &&
```

Comment:
I am simple programmer. `!= null` is good enough for me

Reply:
`is not null`